internal/controllers/machine/machine_controller_test.go

```diff
@@ -2556,6 +2568,7 @@ func TestIsDeleteNodeAllowed(t *testing.T) {
 				},
 				Status: clusterv1.MachineStatus{},
 			},
+			infraMachine:  &InfrastructureMachine{},
```

Comment:
```suggestion
			infraMachine:  nil,
```

Should result in the same right?